sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/RocksDBStateStoreSuite.scala

```diff
@@ -918,6 +918,297 @@ class RocksDBStateStoreSuite extends StateStoreSuiteBase[RocksDBStateStoreProvid
     }
   }
 
+  /* Column family related tests */
+  testWithColumnFamilies("column family creation with invalid names",
```

Comment:
The followings are moved from `RocksDBSuite`. The suites are mostly the same as the old. The only difference is we are testing on the provider layer instead of rocksDB instance layer.